storage/pebble/registers.go

```diff
@@ -61,7 +88,15 @@ func (s *Registers) Store(
 	height uint64,
 	entries flow.RegisterEntries,
 ) error {
-	//  TODO: add value check once LatestHeight is implemented
+	if height == s.latestHeight {
```

Comment:
Do we need to put a mutex on this so we don't mess up the latest height updates?

Reply:
since it's not incremented asynchronously, this should be fine.

Reply:
> since it's not incremented asynchronously, this should be fine.

are you saying `Store` must not be used concurrently?

Reply:
why do we treat this height differently?

Reply:
I was under the impression that it's not: https://github.com/onflow/flow-go/pull/4714#discussion_r1326192014

but I'll implement the mutex guard, I think if we're using it to serve `LatestHeightKey ` and `FirstHeightKey` we should do it anyway.

Reply:
It doesn't make sense for `Store` to be called concurrently, because the register store require the height to be indexed sequentially. You should not index block 11 before block 10 is indexed. That's why we call lock.Lock() just to be safe and preventing any concurrent call. For additional safety, we also require height must be s.latestHeight + 1. 

But what about the case height == s.latestHeight? This probably won't happen for access node, but might happen for execution node, because when execution node is saving a computation result, it not only store the registers in pebble, but also saves events, results, receipts etc to badger (another DB), which means it's a two-phase commit. In the case where saving registers succeeded, but saving the rest data to badger was interrupted by a restart, then on startup, the node will re-execute this block, which will have to store the registers again, in other words, it will call Store with the same height as latestHeight. That's why we can simply skip it, and make the call idempotent.

Does it make sense?